airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/write/load/SnowflakeRecordFormatter.kt

```diff
@@ -8,99 +8,64 @@ import io.airbyte.cdk.load.data.AirbyteValue
+import io.airbyte.cdk.load.schema.model.ColumnSchema
 import io.airbyte.cdk.load.util.Jsons
-import io.airbyte.integrations.destination.snowflake.db.toSnowflakeCompatibleName
-import io.airbyte.integrations.destination.snowflake.sql.SnowflakeColumnUtils
 
 interface SnowflakeRecordFormatter {
     fun format(record: Map<String, AirbyteValue>): List<Any>
 }
 
 class SnowflakeSchemaRecordFormatter(
-    private val columns: LinkedHashMap<String, String>,
-    val snowflakeColumnUtils: SnowflakeColumnUtils,
+    val columnSchema: ColumnSchema,
 ) : SnowflakeRecordFormatter {
+    val userColumns =
+        columnSchema.finalSchema.keys.filterNot { COLUMN_NAMES.contains(it.lowercase()) }
 
-    private val airbyteColumnNames =
-        snowflakeColumnUtils.getFormattedDefaultColumnNames(false).toSet()
+    override fun format(record: Map<String, AirbyteValue>): List<Any> {
+        val result = mutableListOf<Any>()
 
-    override fun format(record: Map<String, AirbyteValue>): List<Any> =
-        columns.map { (columnName, _) ->
-            /*
-             * Meta columns are forced to uppercase for backwards compatibility with previous
-             * versions of the destination.  Therefore, convert the column to lowercase so
-             * that it can match the constants, which use the lowercase version of the meta
-             * column names.
-             */
-            if (airbyteColumnNames.contains(columnName)) {
-                record[columnName.lowercase()].toCsvValue()
-            } else {
-                record.keys
-                    // The columns retrieved from Snowflake do not have any escaping applied.
-                    // Therefore, re-apply the compatible name escaping to the name of the
-                    // columns retrieved from Snowflake.  The record keys should already have
-                    // been escaped by the CDK before arriving at the aggregate, so no need
-                    // to escape again here.
-                    .find { it == columnName.toSnowflakeCompatibleName() }
-                    ?.let { record[it].toCsvValue() }
-                    ?: ""
-            }
-        }
-}
+        // WARNING: Unlike the user fields, the meta fields on the record are not munged for
+        // the destination. So we add meta columns in order using the original lowercase meta key.
+        result.add(record[COLUMN_NAME_AB_RAW_ID].toCsvValue())
+        result.add(record[COLUMN_NAME_AB_EXTRACTED_AT].toCsvValue())
+        result.add(record[COLUMN_NAME_AB_META].toCsvValue())
+        result.add(record[COLUMN_NAME_AB_GENERATION_ID].toCsvValue())
 
-class SnowflakeRawRecordFormatter(
-    columns: LinkedHashMap<String, String>,
-    val snowflakeColumnUtils: SnowflakeColumnUtils,
-) : SnowflakeRecordFormatter {
-    private val columns = columns.keys
+        // Add user columns from the final schema
+        userColumns.forEach { columnName -> result.add(record[columnName].toCsvValue()) }
```

Comment:
why are we able to remove the old `.find { ... }` thing? 

(is this b/c the aggregate is now using the same column names list that we have here, where previously our list was discovered from snowflake, and therefore not the same as what the aggregate was doing)

Reply:
The columns are now properly munged on both the schema and the incoming record